osc-server/src/test/java/org/osc/core/broker/service/tasks/conformance/openstack/securitygroup/SecurityGroupMemberHookCheckTaskTestData.java

```diff
@@ -0,0 +1,212 @@
+import org.osc.core.broker.model.entities.virtualization.ServiceFunctionChain;
+import org.osc.core.broker.model.entities.virtualization.VirtualizationConnector;
+import org.osc.core.broker.model.entities.virtualization.openstack.VM;
+import org.osc.core.broker.model.entities.virtualization.openstack.VMPort;
+import org.osc.core.broker.model.plugin.ApiFactoryService;
+import org.osc.core.broker.service.tasks.conformance.openstack.sfc.SfcFlowClassifierCreateTask;
+import org.osc.core.broker.service.tasks.conformance.openstack.sfc.SfcFlowClassifierDeleteTask;
+import org.osc.core.broker.service.tasks.conformance.openstack.sfc.SfcFlowClassifierUpdateTask;
+import org.osc.sdk.controller.api.SdnRedirectionApi;
+import org.osc.sdk.controller.element.InspectionHookElement;
+import org.osc.sdk.controller.element.InspectionPortElement;
+
+class SecurityGroupMemberHookCheckTaskTestData {
+    private static final String OPENSTACK_SFC_ID = "openstack_sfc_id";
+
+    public static final long SGM_ID = 1L;
+
+    public static final String OPENSTACK_PORT_CLASSIFIER = "openstack_port_classifier";
+
+    public static final String NEUTRON_SFC_CONTROLLER = "Neutron-SFC";
+
+    private ApiFactoryService apiFactoryServiceMock = mock(ApiFactoryService.class);
+    private SdnRedirectionApi redirApiMock = mock(SdnRedirectionApi.class);
+
+    public static SecurityGroupMember SFC_SGM_WITH_NO_PORTS = createBindedSfcSecurityGroupMember(new HashSet<>());
+    public static SecurityGroupMember SFC_SGM_WITH_PORTS = createBindedSfcSecurityGroupMember(getSfcPorts(false));
+    public static SecurityGroupMember SFC_SGM_WITH_DELETED_PORTS = createBindedSfcSecurityGroupMember(
+            getSfcPorts(true));
+    public static SecurityGroupMember SFC_SGM_WITH_MIXED_PORTS = createBindedSfcSecurityGroupMember(
+            getMixedMarkedForDeletedSfcPorts());
+    public static SecurityGroupMember SFC_SGM_WITH_PORTS_SFC_MISMATCH = createBindedSfcSecurityGroupMember(
+            getSfcPorts(false));
+
+    public static SecurityGroupMember SFC_SGM_UNBINDED_WITH_PORTS = createUnBindedSfcSecurityGroupMember(
+            getSfcPorts(false));
+
+    public static SecurityGroupMember SFC_SGM_UNBINDED_WITH_PORTS_UNBINDED = createUnBindedSfcSecurityGroupMember(
+            getSfcPorts(false, false));
+
+    public TaskGraph getSfcWithPortsTaskGraph(SecurityGroupMember sgm) {
+        TaskGraph expectedGraph = new TaskGraph();
+
+        try {
+            initializeApiFactory(sgm);
+            when(this.redirApiMock.getInspectionHook(any())).thenReturn(null);
+        } catch (Exception e) {
+            throw new RuntimeException();
+        }
+        for (VMPort port : sgm.getVmPorts()) {
+            expectedGraph.appendTask(new SfcFlowClassifierCreateTask().create(sgm.getSecurityGroup(), port));
+        }
+        return expectedGraph;
+    }
+
+    public TaskGraph getSfcWithDeletedPortsTaskGraph(SecurityGroupMember sgm) {
```

Comment:
can these graph methods be static?

Reply:
Thats what i tried initially but the api factory and sdn api being shared was causing interference and test were failing

Reply:
I see, can you put a TODO on that so we can take a look later? Maybe something we can improve also on the overall test approach.

Reply:
will add it as part of next commit